### PR TITLE
Support 2DC/NDC compatibility

### PR DESCRIPTION
**Pull request by yux0**
Support 2DC/NDC compatibility

### CODE REVIEW

File: service/history/historyReplicator.go [hunk 1]
```diff
@@ -135,8 +135,8 @@ func newHistoryReplicator(
 				shard,
 				logger,
 				msBuilder,
-				func(mutableState mutableState) mutableStateTaskGenerator {
-					return newMutableStateTaskGenerator(shard.GetDomainCache(), logger, mutableState)
+				func(msBuilder mutableState) mutableStateTaskGenerator {
+					return newMutableStateTaskGenerator(shard.GetDomainCache(), logger, msBuilder)
```
Comment (wxing1292) [inline service/history/historyReplicator.go]: revert this change?

File: service/history/historyReplicator.go [hunk 2]
```diff
@@ -947,14 +947,17 @@ func (r *historyReplicator) terminateWorkflow(
 					domainID,
 					workflowID,
 					runID,
-					msBuilder.GetNextEventID(),
+					mutableState.GetNextEventID(),
 				)
 			}
 
 			// setting the current version to be the last write version
-			msBuilder.UpdateReplicationStateVersion(currentLastWriteVersion, true)
```
Comment (wxing1292) [inline service/history/historyReplicator.go]: all `UpdateReplicationStateVersion` in this file may need to be changed to `UpdateCurrentVersion`

File: service/history/historyReplicator.go (hunk 1 above)
Comment (yux0) [reply service/history/historyReplicator.go]: reverted

File: service/history/historyReplicator.go (hunk 2 above)
Comment (yux0) [reply service/history/historyReplicator.go]: replaced
